### PR TITLE
whitespace-before-parameters comment

### PR DESCRIPTION
**Pull request by arembridge**
**Summary**

Updated doc comment for `whitespace_before_parameters.rs`. Online docs also benefit from this update.

**Test Plan**

Checked docs via [mkdocs](https://github.com/astral-sh/ruff/blob/389fe13c934fe73679474006412b1eded4a2cad0/CONTRIBUTING.md?plain=1#L267-L296)

### CODE REVIEW

File: crates/ruff/src/rules/pycodestyle/rules/logical_lines/whitespace_before_parameters.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use crate::checkers::logical_lines::LogicalLinesContext;
 use crate::rules::pycodestyle::rules::logical_lines::LogicalLine;
 
 /// ## What it does
-/// Checks for extraneous whitespace immediately before an open parenthesis
+/// Checks for extraneous whitespace immediately after an open parenthesis
```
Comment (arembridge) [inline crates/ruff/src/rules/pycodestyle/rules/logical_lines/whitespace_before_parameters.rs:11]: hi @charliermarsh.  

**TLDR:** i've checked and double checked this, but i'm pretty sure that this should be `before` (not `after`) baed on the behaviour of the rule (arguably the rule name is wrong)

I created two test files and executed `cargo run -p ruff_cli -- check <file> --no-cache --select=E211` on each
```python
print ("<- whitespace before parenthesis")
```
produces 
```bash
<path-to-folder>/whitespace_before_parenthesis.py:1:6: E211 [*] Whitespace before '('
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```
while
```python
print( "<- whitespace after parenthesis")
```
produces no output.

Looking at the test file also confirms the `before` behaviour: [not okay](https://github.com/astral-sh/ruff/blob/86539c1fc5afa2516099bf20cdb4fdd85827274a/crates/ruff/resources/test/fixtures/pycodestyle/E21.py#L2)  [okay](https://github.com/astral-sh/ruff/blob/86539c1fc5afa2516099bf20cdb4fdd85827274a/crates/ruff/resources/test/fixtures/pycodestyle/E21.py#L9C1-L9C1) (note there is a space after the `]`)

Arguably, this rule should be called `whitespace_before_parentheses` similar to its documentation on flake8 [here](https://www.flake8rules.com/rules/E211.html)

Let me know if i've missed something! thanks 😊

Comment (charliermarsh) [reply crates/ruff/src/rules/pycodestyle/rules/logical_lines/whitespace_before_parameters.rs:11]: Gah sorry, I will double-check this today. It's possible that any of the rule name, the docs, or the implementation are wrong. Thank you for following up!

Comment (arembridge) [reply crates/ruff/src/rules/pycodestyle/rules/logical_lines/whitespace_before_parameters.rs:11]: i think the answer is to rename the rule.  i can raise a PR for this if you like 😊

Comment (charliermarsh) [reply crates/ruff/src/rules/pycodestyle/rules/logical_lines/whitespace_before_parameters.rs:11]: Thanks again, fixed here: https://github.com/astral-sh/ruff/pull/6133.
